### PR TITLE
Clientside dev

### PR DESCRIPTION
**Pull request by ashish-compro**
Code Review feedback incorporated.

### CODE REVIEW

File: client/app/shared/input-factory.service.ts [hunk 1]
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
...
 
     // create component without adding it directly to the DOM
-    let comp = factory.create(injector);
+    // Creates a new component and returns the ComponentRef
+    let comp = component.create(injector);
 
-    // add inputs first !! otherwise component/template crashes ..
-    comp.instance["data"] = modelInput;
+    // add Data to the instance of the Component.
+    comp.instance["compConfig"] = itemConfig;
```
Comment (dhirendra-singh81) [inline client/app/shared/input-factory.service.ts]: Add a setter fn here

File: client/app/shared/panel/panel.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    this.factoryRef = injector.get(InputFactoryService);
+    this.$el = jQuery(el.nativeElement);
+    jQuery.fn.widgster.Constructor.DEFAULTS.bodySelector = '.widget-body';
+  }
+
+  ngOnInit() {
+    this.labelConfig.rendererProperties.text = this.compConfig.label;
+    this.labelConfig.rendererProperties.type = 'PanelHeading';
+    this.$el.find('.widget').widgster();
+    for (let item of this.compConfig.items) {
```
Comment (dhirendra-singh81) [inline client/app/shared/panel/panel.component.ts]: Put it in a function or maybe we define an interface for a non-leaf component

**Review by dhirendra-singh81 — CHANGES_REQUESTED** [review]
1. Add non-leaf and derive appropriate component from it 
2. Add a setter fn in input factory service comp

File: client/app/shared/input-factory.service.ts [hunk 3]
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
 
   constructor(private componentFactoryResolver: ComponentFactoryResolver) { }
 
-  public createComp (vCref: ViewContainerRef, modelInput){
-     let factory = this.componentFactoryResolver.resolveComponentFactory(this.DynamicCompMap(modelInput.itemRenderer));
+  public createComp (vCref: ViewContainerRef, itemConfig){
```
Comment (vsingb3) [inline client/app/shared/input-factory.service.ts]: vCRef

File: client/app/shared/input-factory.service.ts [hunk 4]
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
...
 
+    /**
+     * [resolveComponentFactory - We create a factory out of the component we want to create]
```
Comment (vsingb3) [inline client/app/shared/input-factory.service.ts]: Get the factory object to create the component

File: client/app/shared/select/select.component.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<div><app-label [compConfig]="labelConfig"></app-label></div>
```
Comment (vsingb3) [inline client/app/shared/select/select.component.html]: Id should be unique for every instance of this component.

Name should be generic

**Review by dhirendra-singh81 — APPROVED** [review]

**Review by dhirendra-singh81 — CHANGES_REQUESTED** [review]
Unapproving

**Review by dhirendra-singh81 — APPROVED** [review]
Logically reviewed. Code refinements still needed.

**Pull request merged**